src/main/java/at/hannibal2/skyhanni/data/QuiverAPI.kt

```diff
@@ -0,0 +1,254 @@
+    )
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        val message = event.message.trimWhiteSpaceAndResets().removeResets()
+
+        selectPattern.matchMatcher(message) {
+            val arrow = group("arrow")
+            currentArrow = getByNameOrNull(arrow) ?: return
+            currentAmount = arrowAmount[currentArrow]?.toInt() ?: 0
+
+            return saveArrowType()
+        }
+
+        fillUpJaxPattern.matchMatcher(message) {
+            val type = group("type")
+            val amount = group("amount").formatNumber().toFloat()
+
+            val filledUpType = getByNameOrNull(type) ?: return
+
+            val existingAmount = arrowAmount[filledUpType] ?: 0f
+            val newAmount = existingAmount + amount
+            arrowAmount[filledUpType] = newAmount
+
+            return saveArrowAmount()
+        }
+
+        fillUpPattern.matchMatcher(message) {
+            val flintAmount = group("flintAmount").formatNumber().toFloat()
+            val existingAmount = arrowAmount[getByNameOrNull("ARROW".asInternalName())] ?: 0f
+            val newAmount = existingAmount + flintAmount
+
+            arrowAmount[getByNameOrFlint("ARROW".asInternalName())] = newAmount
+
+            return saveArrowAmount()
+        }
+
+        addedToQuiverPattern.matchMatcher(message) {
+            val type = group("type")
+            val amount = group("amount").formatNumber().toFloat()
+
+            val filledUpType = getByNameOrNull(type) ?: return
+
+            val existingAmount = arrowAmount[filledUpType] ?: 0f
+            val newAmount = existingAmount + amount
+            arrowAmount[filledUpType] = newAmount
+
+            return saveArrowAmount()
+        }
+
+        clearedPattern.matchMatcher(message) {
+            currentAmount = 0
+            arrowAmount.clear()
+
+            return saveArrowAmount()
+        }
+
+        arrowResetPattern.matchMatcher(message) {
+            currentArrow = getByNameOrNull("NONE".asInternalName()) ?: return
+            currentAmount = 0
+
+            saveArrowType()
+            return saveArrowAmount()
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryFullyLoaded(event: InventoryFullyOpenedEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (event.inventoryName != "Quiver") return
+
+        // clear to prevent duplicates
+        currentAmount = 0
+        arrowAmount.clear()
+
+        val stacks = event.inventoryItems
+        for (stack in stacks.values) {
+            val lore = stack.getLore()
+            if (lore.isEmpty()) continue
+
+            val arrow = stack.getInternalNameOrNull() ?: continue
+
+            val arrowType = getByNameOrNull(arrow) ?: continue
+            val arrowAmount = stack.stackSize + (this.arrowAmount[arrowType] ?: 0.0f)
+
+            this.arrowAmount[arrowType] = arrowAmount
+        }
+
+        saveArrowAmount()
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
```

Comment:
first thing you should check since it is easy to check and very common to fail (any other sound)
```suggestion
    @SubscribeEvent(priority = EventPriority.HIGHEST)
    if(event.soundName != "random.bow") return
```